immuni_exposure_ingestion/tasks/process_uploads_eu.py

```diff
@@ -0,0 +1,146 @@
+#    Copyright (C) 2020 Presidenza del Consiglio dei Ministri.
+#    Please refer to the AUTHORS file for more information.
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+#    GNU Affero General Public License for more details.
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+import asyncio
+import logging
+from datetime import datetime
+from typing import List
+
+from bson import ObjectId
+from croniter import croniter
+
+from immuni_common.models.mongoengine.batch_file_eu import BatchFileEu
+from immuni_common.models.mongoengine.temporary_exposure_key import TemporaryExposureKey
+from immuni_exposure_ingestion.celery import celery_app
+from immuni_exposure_ingestion.core import config
+from immuni_exposure_ingestion.helpers.lock import lock_concurrency
+from immuni_exposure_ingestion.helpers.risk_level import set_highest_risk_level_from_upload
+from immuni_exposure_ingestion.models.upload_eu import UploadEu
+from immuni_exposure_ingestion.monitoring.celery import (
+    BATCH_FILES_EU_CREATED,
+    KEYS_EU_PROCESSED,
+    UPLOADS_EU_ENQUEUED,
+)
+from immuni_exposure_ingestion.protobuf.helpers.generate_zip import batch_to_sdk_zip_file
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@celery_app.task()
+def process_uploads_eu() -> None:
+    """
+    Celery does not support async functions, so we wrap it around asyncio.run.
+    """
+    asyncio.run(_process_uploads_eu())  # pragma: no cover
+
+
+async def _process_uploads_eu() -> None:
+    """
+    This task is run every "PERIOD".
+    The period is defined in config, it can be thought of as a period of 6 / 12 / 24 hours.
+
+    The task will create a batch file from the unprocessed Uploads of the same period, and perform
+    some minor validations.
+
+    If the number of keys in the given uploads is greater than the maximum number of keys allowed in
+    a single batch, the task will create multiple batches and group them into "sub-batches" instead.
```

Comment:
I'd change this sentence, also in the original one, as it doesn't hold given the current implementation.